RentACarWebAPI/Services/BaseService.cs

```diff
@@ -0,0 +1,47 @@
+﻿using RentACarWebAPI.Interfaces.Repositories;
+using RentACarWebAPI.Interfaces.Services;
+using RentACarWebAPI.Models.Base;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace RentACarWebAPI.Services
+{
+    public class BaseService<T> : IBaseRepository<T> where T : Entity
+    {
+        private readonly IBaseRepository<T> _entityRepository;
+        public BaseService(IBaseRepository<T> entityRepository )
+        {
+            _entityRepository = entityRepository;
+        }
```

Comment:
Super minor but don't forget to use empty lines. For example it would be nice to add one here and between constructor & private readonly instance.

Reply:
You are right! my brother has told me that since I was 9 years old 🤣 and I always forget from somes! I will pay more attention! thanks!!

Reply:
oh nice! good advice! thanks!